app/logic/Mail/ActiveSync/ActiveSyncAccount.ts

```diff
@@ -117,6 +117,20 @@ export class ActiveSyncAccount extends MailAccount {
     if (email.bcc.hasItems) {
       throw new NotSupported("ActiveSync does not support BCC");
     }
+    assert(email.folder?.id, "Need folder to save the sent email in");
+    let defaultSentFolder = this.getSpecialFolder(SpecialFolder.Sent) as ActiveSyncFolder;
+    if (email.folder.specialFolder != SpecialFolder.Sent) {
+      await defaultSentFolder.listMessages();
+    }
+    defaultSentFolder.newMessageCallbacks.add(message => {
```

Comment:
Can we do that only when the sent folder is not the default ActiveSync sent folder? I.e. within the `if` block above?

Reply:
The idea is that we track all sent messages so that we can explicitly ignore the ones that shouldn't be moved.

Reply:
What's the advantage? Why not avoid adding the listener at all in this case? We already know at the time before we add the listener.

I want to avoid running this code altogether in the normal case, and do this special stuff only when the user explicitly sets the sent folder to a non-default folder.

Reply:
Sorry, I don't understand. If we do *not* need to move a message, i.e. there's nothing to do, why would we add a callback?